catalogs/catalog-hive/src/main/java/org/apache/gravitino/catalog/hive/HiveCatalogOperations.java

```diff
@@ -539,23 +538,32 @@ public NameIdentifier[] listTables(Namespace namespace) throws NoSuchSchemaExcep
       // then based on
       // those names we can obtain metadata for each individual table and get the type we needed.
       List<String> allTables = clientPool.run(c -> c.getAllTables(schemaIdent.name()));
-      return clientPool.run(
-          c ->
-              c.getTableObjectsByName(schemaIdent.name(), allTables).stream()
-                  .filter(
-                      tb -> {
-                        boolean isSupportTable = SUPPORT_TABLE_TYPES.contains(tb.getTableType());
-                        if (!isSupportTable) {
-                          return false;
-                        }
-                        if (!listAllTables) {
-                          Map<String, String> parameters = tb.getParameters();
-                          return isHiveTable(parameters);
-                        }
-                        return true;
-                      })
-                  .map(tb -> NameIdentifier.of(namespace, tb.getTableName()))
-                  .toArray(NameIdentifier[]::new));
+      if (!listAllTables) {
```

Comment:
@mchades I think you should also check Paimon catalog, AFAIK, I have supported Paimon with HMS backend recently. Please sync with @FANNG1 and @caican00 .

Reply:
Based on [the source codes of Paimon ](https://github.com/apache/paimon/blob/76f906faad4a155f75f6db52a5c0014d532b2c55/paimon-hive/paimon-hive-catalog/src/main/java/org/apache/paimon/hive/HiveCatalog.java#L1021)and tested locally, I updated the PR and use table param `table_type=paimon` to filter out Paimon tables.